snorkel/labeling/model/wrapper.py

```diff
@@ -0,0 +1,191 @@
+from typing import Any, List, Optional
+
+import numpy as np
+from sklearn.base import BaseEstimator, ClassifierMixin  # type: ignore
+
+from snorkel.labeling import LabelModel
+from snorkel.utils.lr_schedulers import LRSchedulerConfig
+from snorkel.utils.optimizers import OptimizerConfig
+
+
+class SKlearnLabelModel(BaseEstimator, ClassifierMixin):
+    """A sklearn wrapper for LabelModel for using sklearn GridSearch and RandomSearch.
+
+    Note that all hyperparameters for the fit and score functions are accepted at the time the class is defined.
+    This includes L_train.
+
+    **NOTE: you have to set cv=[(slice(None), slice(None))] when you class scikit-learn search to make sure to validate on L_dev, not L_train**
+
+    Examples
+    --------
+    >>> L_train = np.array([[1, 1, 1, 0], [0, 1, 1, 0], [1, 0, 0, 1], [0, 1, 0, 1], [0, 0, 0, 1]])
+    >>> L_dev = np.array([[1, 1, 1, 0], [0, 1, 1, 0], [1, 0, 0, 1], [0, 1, 0, 1]])
+    >>> Y_dev = np.array([1, 1, 0, 1])
+
+    >>> from sklearn.model_selection import GridSearchCV
+    >>> param_grid = [{"lr": [0.01, 0.0000001], "l2": [0.001, 0.5], "metric": ["accuracy"]}]
+    >>> clf = GridSearchCV(SKlearnLabelModel(L_train), param_grid, cv=[(slice(None), slice(None))])
```

Comment:
`(slice(None), slice(None))` and passing L_train to init is a code smell, but grrrrr all of sklearn's model selection utils are all tied to CV, not generic validation set fitting. did you look into `PredefinedSplit`? we could provide a utility that glues L_train and L_dev together, makes an appropriate Y, then produces the corresponding `PredefinedSplit` object for people to use with their *CV calls.

Reply:
Note to self: https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.PredefinedSplit.html

Setting all of L_train’s indices to -1 in test split should fix the issue. Gridsearch complains when number of splits is less than 1, but will test wrapping in PredefinedSplit